plugins/plugin-ssh-machine/src/main/java/org/eclipse/che/plugin/machine/ssh/SshMachineImplTerminalLauncher.java

```diff
@@ -77,7 +81,10 @@ public void launch(Instance machine, Agent agent) throws ServerException {
             String architecture = detectArchitecture(machine);
             machine.copy(archivePathProvider.getPath(architecture), terminalLocation);
 
-            super.launch(machine, agent);
+            final AgentImpl agentCopy = new AgentImpl(agent);
```

Comment:
Do you see the necessity to extend `TerminalAgentLuncher` directly to avoid copying of the code?
Moreover you can get rid of the `getAgentId()` method.
@tolusha, @svor